repeated-integration-tests.sh

```diff
@@ -15,22 +15,32 @@ function cleanup() {
 trap cleanup EXIT
 
 
+FLAGS=(
+    "exec:exec@create-pod"
+    "exec:exec@start-jfr-datasource"
+    "exec:exec@start-grafana"
+    "exec:exec@start-container"
+    "exec:exec@wait-for-container"
+    "failsafe:integration-test"
+    "exec:exec@stop-jfr-datasource"
+    "exec:exec@stop-grafana"
+    "exec:exec@stop-container"
+    "exec:exec@destroy-pod"
+)
+
+if command -v ansi2txt; then
+    FLAGS+=("-Dstyle.color=always")
+    PIPECLEANER=ansi2txt
+else
+    PIPECLEANER=cat
+fi
+
 runcount=0
 while [ "${runcount}" -lt "${runs}" ]; do
     timestamp="$(date -Iminutes)"
     client_logfile="cryostat-itests-${timestamp}.client.log"
     server_logfile="cryostat-itests-${timestamp}.server.log"
-    mvn \
-        exec:exec@create-pod \
-        exec:exec@start-jfr-datasource \
-        exec:exec@start-grafana \
-        exec:exec@start-container \
-        exec:exec@wait-for-container \
-        failsafe:integration-test \
-        exec:exec@stop-jfr-datasource \
-        exec:exec@stop-grafana \
-        exec:exec@stop-container \
-        exec:exec@destroy-pod |& tee -a "${client_logfile}"
+    mvn "${FLAGS[@]}" |& tee >($PIPECLEANER > ${client_logfile})
```

Comment:
Does this error show up for you or is it just me?
```
repeated-integration-tests.sh: line 43: syntax error near unexpected token `('
repeated-integration-tests.sh: line 43: `    mvn "${FLAGS[@]}" |& tee >($PIPECLEANER > ${client_logfile})'
```

Reply:
I don't get that. That syntax is bash-specific - what are you using? The shebang line at the top of the file should cause it to be run with bash, but maybe not in this case.

Reply:
Ah, running `bash repeated-integration-tests.sh` works fine, but `sh repeated-integration-tests.sh` results in the syntax error. Interesting

Reply:
Odd. What's `which sh` or `sh --version` say?

Reply:
Or most tellingly, `ls -l $(which sh)`

Reply:
```
[jalaw@localhost cryostat]$ ls -l $(which sh)
lrwxrwxrwx. 1 root root 4 Jul 27  2020 /usr/bin/sh -> bash

[jalaw@localhost cryostat]$ sh --version
GNU bash, version 5.0.17(1)-release (x86_64-redhat-linux-gnu)
```

Reply:
Very weird.

```bash
~ $ ls -l $(which sh)
lrwxrwxrwx. 1 root root 4 Jan 25  2021 /usr/bin/sh -> bash

~ $ sh --version
GNU bash, version 5.1.0(1)-release (x86_64-redhat-linux-gnu)
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

Reply:
Here's the latest:
```
jalaw@localhost ~/cryostat (repeated-itest-script-fixup)> git pull
Updating 3d521e26..3b553584
Fast-forward
 repeated-integration-tests.sh => repeated-integration-tests.bash | 11 +++++++----
 repeated-unit-tests.bash                                         | 38 ++++++++++++++++++++++++++++++++++++++
 repeated-unit-tests.sh                                           | 26 --------------------------
 3 files changed, 45 insertions(+), 30 deletions(-)
 rename repeated-integration-tests.sh => repeated-integration-tests.bash (83%)
 create mode 100755 repeated-unit-tests.bash
 delete mode 100755 repeated-unit-tests.sh
jalaw@localhost ~/cryostat (repeated-itest-script-fixup)> sh repeated-integration-tests.bash 
repeated-integration-tests.bash: line 45: syntax error near unexpected token `('
repeated-integration-tests.bash: line 45: `    mvn "${FLAGS[@]}" |& tee >($PIPECLEANER > "${client_logfile}")'
jalaw@localhost ~/cryostat (repeated-itest-script-fixup) [1]> bash repeated-integration-tests.bash 
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------------< io.cryostat:cryostat >------------------------
[INFO] Building cryostat 2.1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:3.0.0:exec (create-pod) @ cryostat ---
...
```

Reply:
And for me, all of `sh repeated-integration-tests.bash` (no tab complete on this one), `bash repeated-integration-tests.bash`, and `./repeated-integration-tests.bash` all seem to work fine.

`echo $SHELL`? `/bin/bash` for me.

Reply:
> Here's the latest:

Not really sure - I guess just start explicitly calling it with `bash` then. I could rewrite that part of the script to make it plain `sh`-compliant but there are several overall `bash`isms in there and it'd be a bit annoying to do.